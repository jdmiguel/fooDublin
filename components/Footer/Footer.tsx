import React from 'react';
import styled from 'styled-components';

type FooterProps = {
  isExtended: boolean;
  onClickFavourites?: () => void;
};

const StyledFooterWrapper = styled.footer`
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
`;

const StyledNavFooterWrapper = styled.div`
  border-top: 1px solid ${(props) => props.theme.palette.LIGHT_SOFT};
  padding: 15px 20px;
`;

const StyledNavFooter = styled.div`
  display: flex;
  justify-content: center;
  margin: 0 auto;
  max-width: 1100px;
  @media only screen and (min-width: 768px) {
    justify-content: space-between;
  }
`;

const StyledFavLink = styled.button`
  color: ${(props) => props.theme.palette.PRIMARY};
  font-weight: 600;
  transition: color 0.2s ease-out;
  cursor: pointer;
  background: none;
  outline: none;
  display: none;
  i {
    font-size: 0.9rem;
    margin-right: 4px;
  }
  &:hover {
    color: ${(props) => props.theme.palette.PRIMARY_DARK};
  }
  @media only screen and (min-width: 768px) {
    display: flex;
    align-items: center;
  }
`;

const StyledRightsFooter = styled.div`
  background-color: ${(props) => props.theme.palette.SECONDARY};
  display: flex;
  justify-content: center;
  padding: 20px 0;
`;

const StyledText = styled.p`
  color: ${(props) => props.theme.palette.LIGHT_SOFT};
  font-weight: 300;
  font-size: 0.9em;
  &:not(:first-of-type) {
    margin-left: 5px;
    @media only screen and (min-width: 375px) {
      margin-left: 7px;
    }
  }
  &:not(:last-of-type) {
    &:after {
      content: ' | ';
      margin-left: 3px;
      @media only screen and (min-width: 375px) {
        margin-left: 5px;
      }
    }
  }
  &:last-of-type {
    margin-right: 1px;
    @media only screen and (min-width: 375px) {
      margin-right: 3px;
    }
  }
  @media only screen and (min-width: 375px) {
    font-size: 1em;
  }
`;

const StyledLink = styled.a`
  color: ${(props) => props.theme.palette.LIGHT_SOFT};
  font-weight: 600;
  font-size: 0.9em;
  text-decoration: none;
  transition: color 0.2s ease-out;
  &:hover {
    color: ${(props) => props.theme.palette.LIGHT_MAX};
  }
  @media only screen and (min-width: 375px) {
    font-size: 1em;
  }
`;

const Footer: React.FC<FooterProps> = ({ isExtended, onClickFavourites }) => (
  <StyledFooterWrapper>
    {isExtended && (
      <StyledNavFooterWrapper>
        <StyledNavFooter>
          <div>Home / Search Fast Food in Rathmines / Elephant & Castle</div>
          <StyledFavLink
            onClick={(event: React.MouseEvent<HTMLElement>) => {
              event.preventDefault();
              onClickFavourites && onClickFavourites();
            }}
          >
            <i className="material-icons">bookmarks</i>FAVOURITES
          </StyledFavLink>
        </StyledNavFooter>
      </StyledNavFooterWrapper>
    )}
    <StyledRightsFooter>
      <StyledText>FOODUBLIN</StyledText>
      <StyledText>COPYRIGHT Â©2020</StyledText>
      <StyledText>BY</StyledText>
      <StyledLink href="https://jdmiguel.netlify.app/" target="_blank">
        JDMIGUEL
      </StyledLink>
    </StyledRightsFooter>
  </StyledFooterWrapper>
);

export default Footer;
