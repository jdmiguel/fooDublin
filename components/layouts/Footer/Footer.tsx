import { CustomLink } from '../../core/CustomLink/CustomLink';
import {
  StyledFooterWrapper,
  StyledFooterVeil,
  StyledFooter,
  StyledBlock,
  StyledText,
} from './styles';

type FooterProps = {
  showVeil?: boolean;
};

export const Footer: React.FC<FooterProps> = ({ showVeil = false }) => (
  <StyledFooterWrapper data-testid="footer">
    {showVeil && <StyledFooterVeil />}
    <StyledFooter>
      <StyledBlock>
        <StyledText addSeparation={false}>FOODUBLIN Â©2022</StyledText>
      </StyledBlock>
      <StyledBlock>
        <StyledText addSeparation={true}>BY</StyledText>
        <CustomLink route="https://jdmiguel.netlify.app/">JDMIGUEL</CustomLink>
      </StyledBlock>
      <StyledBlock>
        <StyledText addSeparation={false}></StyledText>
        <CustomLink route="https://foodublin-design-system.netlify.app">DESIGN SYSTEM</CustomLink>
      </StyledBlock>
    </StyledFooter>
  </StyledFooterWrapper>
);
