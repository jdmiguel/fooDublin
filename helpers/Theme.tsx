import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#EC7E00',
    PRIMARY_DARK: '#D48C24',
    PRIMARY_MEDIUM: '#FBA52B',
    PRIMARY_LIGHT: '#FDF3E5',
    SECONDARY: '#660000',
    LIGHT_MAX: '#FFFFFF',
    LIGHT_MEDIUM: '#F3F3F3',
    LIGHT_MIN: '#E5EAED',
    LIGHT_SOFT: '#D8D8D8',
    DARK_MAX: '#222C3C',
    DARK_MEDIUM: '#585858',
    DARK_MIN: '#89959B',
    DARK_SOFT: '#A7A7A7',
  },
};

export const renderWithTheme = (component: any) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
