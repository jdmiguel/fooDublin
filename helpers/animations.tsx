import { keyframes, css } from 'styled-components';

const fadeIn = (endValue: number) => keyframes`
  0% { opacity: 0 }
  100% { opacity: ${endValue} }
`;

const fadeInFromPosY = (initialPosY: number) => keyframes`
  0% { opacity: 0; transform: translateY(${initialPosY}px); }
  100% { opacity: 1; ; transform: translateY(0); }
`;

const rotate = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const moveX = keyframes`
  0% { transform: translateX(-100%); }
  100% { transform: translateX(0); }
`;

export const fadeInAnimation = css<{ endValue?: number }>`
  animation: ${({ endValue }) => fadeIn(endValue || 1)} 0.15s linear;
`;

export const fadeInFromPosYAnimation = css<{
  initialPosY?: number;
  animationDuration?: number;
  animationDelay?: number;
}>`
  animation-name: ${({ initialPosY }) => fadeInFromPosY(initialPosY || 50)};
  animation-duration: ${({ animationDuration }) => animationDuration || 0.15}s;
  animation-delay: ${({ animationDelay }) => animationDelay || 0.25}s;
  animation-fill-mode: forwards;
  animation-timing-function: ease-out;
`;

export const circleLoaderAnimation = css`
  animation: ${rotate} 0.8s linear infinite;
`;

export const lineLoaderAnimation = css`
  animation: ${moveX} 1s ease-in-out infinite forwards;
`;
