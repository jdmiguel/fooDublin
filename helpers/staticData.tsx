import { CDN_URL_STATIC_DIRECTORY } from './utils';

export const DUBLIN_ID = 91;
export const BASE_URL = 'https://developers.zomato.com/api/v2.1/';

export const DEFAULT_TEXT_LOADING = 'Coming right up...';

export const MAX_RESTAURANT_DISPLAYED = 20;
export const MAX_RESTAURANT_RETRIEVED = 100;

export const MAX_SMALL_DEVICE_WIDTH = 640;

export const SCROLL_FACTOR = 4;
export const SCROLL_INITIAL_MOBILE_FACTOR = 10;
export const SCROLL_OFFSET_MOBILE_FACTOR = 4;
export const SCROLL_OFFSET_DESKTOP_FACTOR = 1.2;
export const SCROLL_DELAY = 4;

export const SHOWING_SCROLLUP_BUTTON_HEIGHT = 900;

export enum PlaceholderText {
  BLURRED = 'Search for locals...',
  FOCUSED = 'Start typing to search...',
}

export const DEFAULT_SUGGESTIONS = [
  {
    id: 0,
    imgSrc: '',
    firstText: '',
    secondText: '',
  },
];

export const LOCATIONS = [
  {
    id: 91,
    name: 'Dublin',
    path: 'dublin',
  },
  {
    id: 162239,
    name: 'South City West',
    path: 'south-city-west',
  },
  {
    id: 96201,
    name: 'Temple Bar',
    path: 'temple-bar',
  },
  {
    id: 95051,
    name: 'Clondalkin',
    path: 'clondalkin',
  },
  {
    id: 96181,
    name: 'Swords',
    path: 'swords',
  },
  {
    id: 96191,
    name: 'Tallaght',
    path: 'tallaght',
  },
  {
    id: 94881,
    name: 'Blanchardstown',
    path: 'blanchardstown',
  },
  {
    id: 162240,
    name: 'South City East',
    path: 'south-city-east',
  },
  {
    id: 95311,
    name: 'Dundrum',
    path: 'dundrum',
  },
  {
    id: 95341,
    name: 'Finglas',
    path: 'finglas',
  },
  {
    id: 94791,
    name: 'Ballyfermot',
    path: 'ballyfermot',
  },
  {
    id: 96081,
    name: 'Santry',
    path: 'santry',
  },
  {
    id: 95711,
    name: 'Lucan',
    path: 'lucan',
  },
  {
    id: 94671,
    name: 'Artane',
    path: 'artane',
  },
  {
    id: 95721,
    name: 'Malahide',
    path: 'malahide',
  },
  {
    id: 95301,
    name: 'Dun Laoghaire',
    path: 'dun-laoghaire',
  },
  {
    id: 96271,
    name: 'Walkinstown',
    path: 'walkinstown',
  },
  {
    id: 94871,
    name: 'Blackrock',
    path: 'blackrock',
  },
  {
    id: 95851,
    name: 'Palmerstown',
    path: 'palmerstown',
  },
  {
    id: 95981,
    name: 'Rathmines',
    path: 'rathmines',
  },
  {
    id: 94701,
    name: 'Balbriggan',
    path: 'balbriggan',
  },
  {
    id: 94761,
    name: 'Ballybrack',
    path: 'ballybrack',
  },
  {
    id: 96111,
    name: 'Smithfield',
    path: 'smithfield',
  },
  {
    id: 95181,
    name: 'Crumlin',
    path: 'crumlin',
  },
  {
    id: 94971,
    name: 'Castleknock',
    path: 'castleknock',
  },
  {
    id: 95511,
    name: 'Howth',
    path: 'howth',
  },
  {
    id: 94951,
    name: 'Cabra',
    path: 'cabra',
  },
  {
    id: 96291,
    name: 'Whitehall',
    path: 'whitehall',
  },
  {
    id: 95901,
    name: 'Portmarnock',
    path: 'portmarnock',
  },
  {
    id: 95641,
    name: 'Kimmage',
    path: 'kimmage',
  },
];

export const CUISINES = [
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/stew.svg`,
    id: 135,
    name: 'Irish',
    path: 'irish',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/pizza.svg`,
    id: 55,
    name: 'Italian',
    path: 'italian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/bruschetta.svg`,
    id: 70,
    name: 'Mediterranean',
    path: 'mediterranean',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/rice.svg`,
    id: 3,
    name: 'Asian',
    path: 'asian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/healthy_food.svg`,
    id: 143,
    name: 'Healthy Food',
    path: 'healthy-food',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/gulab.svg`,
    id: 148,
    name: 'Indian',
    path: 'indian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/taco.svg`,
    id: 73,
    name: 'Mexican',
    path: 'mexican',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/fries.svg`,
    id: 40,
    name: 'Fast Food',
    path: 'fast-food',
  },
];

export const HIGHLIGHTED_RESTAURANTS = [
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/cleaver-east.jpg`,
    id: 9100233,
    path: 'cleaver-east',
    name: 'Cleaver East Restaurant',
    description:
      'Situated in the heart of Ireland’s capital city, on East Essex Street in the popular Temple Bar area, Cleaver East Restaurant offers the finest Irish and European cuisine.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/tapas-de-lola.jpg`,
    id: 16518539,
    path: 'tapas-de-lola',
    name: 'Tapas de Lola',
    description:
      'This restaurant is a love letter to all things Spanish and flavorful. Its owners make regular pilgrimages to Spain for inspiration and pack it all into a lively menu of small plates.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/ely-wine.jpg`,
    id: 16517305,
    path: 'ely-wine',
    name: 'Ely Wine Bar',
    description:
      'Eric and his wife Michelle are the founders of this clubby place where you can eat and drink well both downstairs in the cozy basement and upstairs in an elegant dining room.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/oxmantown.jpg`,
    id: 16520229,
    path: 'oxmantown',
    name: 'Oxmantown',
    description:
      'In this small place, the art of a brilliant sandwich means every element is taken seriously, like the breakfast sandwich that comes with butcher Jack McCarthy’s black pudding.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/chapter-one.jpg`,
    id: 9100702,
    path: 'chapter-one',
    name: 'Chapter One',
    description:
      'Dublin writers are more closely associated with booze rather than food, but Chapter One is a Michelin-starred restaurant in the basement of the Dublin Writers Museum.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/rosa-madre.jpg`,
    id: 9101166,
    path: 'rosa-madre',
    name: 'Rosa Madre',
    description:
      'In the menu-hawking clamor of Temple Bar you might long for a quieter restaurant where you won’t feel like a tourist on a conveyor belt. If so, Rosa Madre is your place.',
  },
];

export const HIGHLIGHT_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/highlight-generic.png`;
export const THUMB_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/generic-thumb.png`;
export const DETAIL_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/generic-detail.png`;
export const DEFAULT_RATING_STAR_LIST = [
  { id: 1, type: 'star_outline' },
  { id: 2, type: 'star_outline' },
  { id: 3, type: 'star_outline' },
  { id: 4, type: 'star_outline' },
  { id: 5, type: 'star_outline' },
];

export enum FilterSort {
  COST = 'cost',
  RANK = 'rating',
}
export enum FilterOrder {
  ASC = 'asc',
  DESC = 'desc',
}

export const FILTER_DATA = [
  {
    primaryText: 'cost',
    secondaryText: '- high to low',
    icon: 'keyboard_arrow_up',
    sort: FilterSort.COST,
    order: FilterOrder.ASC,
    id: 1,
  },
  {
    primaryText: 'cost',
    secondaryText: '- low to high',
    icon: 'keyboard_arrow_down',
    sort: FilterSort.COST,
    order: FilterOrder.DESC,
    id: 2,
  },
  {
    primaryText: 'rank',
    secondaryText: '- high to low',
    icon: 'keyboard_arrow_up',
    sort: FilterSort.RANK,
    order: FilterOrder.ASC,
    id: 3,
  },
  {
    primaryText: 'rank',
    secondaryText: '- low to high',
    icon: 'keyboard_arrow_down',
    sort: FilterSort.RANK,
    order: FilterOrder.DESC,
    id: 4,
  },
];
