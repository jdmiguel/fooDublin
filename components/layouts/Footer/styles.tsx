import styled, { css } from 'styled-components';
import { fadeInAnimation } from '@/helpers/animations';

const smallDevicesTextCSS = css`
  @media only screen and (max-width: 330px) {
    font-size: 0.85rem;
  }
`;

export const StyledFooterWrapper = styled.footer`
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  overflow: hidden;
  z-index: 1;
`;

export const StyledFooterVeil = styled.div`
  background-color: ${({ theme }) => theme.palette.LIGHT_MEDIUM};
  height: 170px;
  opacity: 0.6;
  ${fadeInAnimation};
  @media only screen and (min-width: 768px) {
    height: 150px;
  }
`;

export const StyledFooter = styled.div`
  background-color: ${({ theme }) => theme.palette.SECONDARY};
  display: flex;
  justify-content: center;
  padding: 20px 0;
`;

export const StyledBlock = styled.div`
  display: flex;
  color: ${({ theme }) => theme.palette.LIGHT_SOFT};
  &:not(:first-of-type) {
    margin-left: 5px;
    @media only screen and (min-width: 375px) {
      margin-left: 7px;
    }
  }
  &:not(:last-of-type) {
    &:after {
      content: ' | ';
      margin-left: 3px;
      @media only screen and (min-width: 375px) {
        margin-left: 5px;
      }
    }
  }
  &:last-of-type {
    margin-right: 1px;
    @media only screen and (min-width: 375px) {
      margin-right: 3px;
    }
  }
  ${smallDevicesTextCSS}
`;

export const StyledText = styled.p<{ addSeparation: boolean }>`
  font-weight: 300;
  margin-right: ${({ addSeparation }) => addSeparation && '3px'};
  ${smallDevicesTextCSS}
`;

export const StyledFooterBarWrapper = styled.div`
  border-top: 1px solid ${({ theme }) => theme.palette.LIGHT_SOFT};
`;

export const StyledFooterBar = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 15px 20px;
  @media only screen and (min-width: 600px) {
    padding: 15px 30px;
  }
  @media only screen and (min-width: 768px) {
    padding: 15px 35px;
  }
  @media only screen and (min-width: 1024px) {
    padding: 15px 48px;
  }
  @media only screen and (min-width: 1200px) {
    padding: 15px 30px;
  }
`;
