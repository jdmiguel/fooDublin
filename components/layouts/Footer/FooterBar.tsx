import React from 'react';
import { useSelector } from 'react-redux';

import { Breadcrumbs } from '../../core/Breadcrumbs/Breadcrumbs';
import { CustomLink } from '../../core/CustomLink/CustomLink';

import { StyledFooterBarWrapper, StyledFooterBar } from './styles';

import { InitialAppState } from '@/store/redux/types';

type FooterBarProps = {
  onClickBreadcrumb: (route: string, asRoute: string) => void;
  onClickFavorites: () => void;
};

export const FooterBar: React.FC<FooterBarProps> = ({
  onClickBreadcrumb,
  onClickFavorites,
}) => {
  const { breadcrumbs } = useSelector((state: InitialAppState) => state);

  return (
    <StyledFooterBarWrapper>
      <StyledFooterBar className="grid-container">
        <Breadcrumbs
          breadcrumbsData={breadcrumbs || []}
          onClickBreadcrumb={onClickBreadcrumb}
        />
        <CustomLink onClick={onClickFavorites}>
          <i className="material-icons">bookmarks</i>FAVORITES
        </CustomLink>
      </StyledFooterBar>
    </StyledFooterBarWrapper>
  );
};
