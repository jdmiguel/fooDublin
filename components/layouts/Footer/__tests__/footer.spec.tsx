/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';
import { Footer } from '../Footer';
import { FOOTER_MOCKS } from '../__mocks__/footer.mocks';
import { renderWithTheme } from '../../../../helpers/Theme';

describe('Component: Footer', () => {
  it('should render', () => {
    const { container } = render(renderWithTheme(<Footer {...FOOTER_MOCKS} />));

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with veil', () => {
    const updateFooterProps = { ...FOOTER_MOCKS, showVeil: true };
    const { container } = render(renderWithTheme(<Footer {...updateFooterProps} />));

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render extended', () => {
    const updateFooterProps = { ...FOOTER_MOCKS, isExtended: true };
    const { container } = render(renderWithTheme(<Footer {...updateFooterProps} />));

    expect(container.firstChild).toMatchSnapshot();
  });
});
