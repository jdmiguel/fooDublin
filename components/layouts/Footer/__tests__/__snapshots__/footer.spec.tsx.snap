// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer should render 1`] = `
.c3 {
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  cursor: pointer;
  color: #FBA52B;
}

.c3 i {
  font-size: 0.8em;
  margin-right: 4px;
}

.c0 {
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  overflow: hidden;
}

.c1 {
  background-color: #660000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 20px 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #D8D8D8;
}

.c2:not(:first-of-type) {
  margin-left: 5px;
}

.c2:not(:last-of-type):after {
  content: ' | ';
  margin-left: 3px;
}

.c2:last-of-type {
  margin-right: 1px;
}

.c5 {
  font-weight: 300;
}

.c6 {
  font-weight: 300;
  margin-right: 3px;
}

@media only screen and (min-width:540px) {
  .c3 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }

  .c3:hover {
    color: #EC7E00;
  }
}

@media only screen and (max-width:330px) {
  .c4 {
    font-size: 0.85rem;
  }
}

@media only screen and (min-width:375px) {
  .c2:not(:first-of-type) {
    margin-left: 7px;
  }
}

@media only screen and (min-width:375px) {
  .c2:not(:last-of-type):after {
    margin-left: 5px;
  }
}

@media only screen and (min-width:375px) {
  .c2:last-of-type {
    margin-right: 3px;
  }
}

@media only screen and (max-width:330px) {
  .c2 {
    font-size: 0.85rem;
  }
}

@media only screen and (max-width:330px) {
  .c5 {
    font-size: 0.85rem;
  }
}

@media only screen and (max-width:330px) {
  .c6 {
    font-size: 0.85rem;
  }
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <a
        class="c3 c4"
        href="https://github.com/jdmiguel/foodublin"
      >
        GITHUB
      </a>
    </div>
    <div
      class="c2"
    >
      <p
        class="c5"
      >
        FOODUBLIN ©2020
      </p>
    </div>
    <div
      class="c2"
    >
      <p
        class="c6"
      >
        BY
      </p>
      <a
        class="c3 c4"
        href="https://jdmiguel.netlify.app/"
      >
        JDMIGUEL
      </a>
    </div>
  </div>
</footer>
`;

exports[`Component: Footer should render the extended version 1`] = `
.c4 {
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  cursor: pointer;
  color: #FBA52B;
}

.c4 i {
  font-size: 0.8em;
  margin-right: 4px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  overflow: hidden;
}

.c1 {
  border-top: 1px solid #D8D8D8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 15px 10px;
}

.c6 {
  background-color: #660000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 20px 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #D8D8D8;
}

.c7:not(:first-of-type) {
  margin-left: 5px;
}

.c7:not(:last-of-type):after {
  content: ' | ';
  margin-left: 3px;
}

.c7:last-of-type {
  margin-right: 1px;
}

.c8 {
  font-weight: 300;
}

.c9 {
  font-weight: 300;
  margin-right: 3px;
}

@media only screen and (min-width:540px) {
  .c4 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }

  .c4:hover {
    color: #EC7E00;
  }
}

@media only screen and (min-width:600px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media only screen and (min-width:600px) {
  .c2 {
    padding: 15px 20px;
  }
}

@media only screen and (max-width:330px) {
  .c5 {
    font-size: 0.85rem;
  }
}

@media only screen and (min-width:375px) {
  .c7:not(:first-of-type) {
    margin-left: 7px;
  }
}

@media only screen and (min-width:375px) {
  .c7:not(:last-of-type):after {
    margin-left: 5px;
  }
}

@media only screen and (min-width:375px) {
  .c7:last-of-type {
    margin-right: 3px;
  }
}

@media only screen and (max-width:330px) {
  .c7 {
    font-size: 0.85rem;
  }
}

@media only screen and (max-width:330px) {
  .c8 {
    font-size: 0.85rem;
  }
}

@media only screen and (max-width:330px) {
  .c9 {
    font-size: 0.85rem;
  }
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 grid-container"
    >
      <div
        class="c3"
      />
      <a
        class="c4 c5"
      >
        <i
          class="material-icons"
        >
          bookmarks
        </i>
        FAVORITES
      </a>
    </div>
  </div>
  <div
    class="c6"
  >
    <div
      class="c7"
    >
      <a
        class="c4 c5"
        href="https://github.com/jdmiguel/foodublin"
      >
        GITHUB
      </a>
    </div>
    <div
      class="c7"
    >
      <p
        class="c8"
      >
        FOODUBLIN ©2020
      </p>
    </div>
    <div
      class="c7"
    >
      <p
        class="c9"
      >
        BY
      </p>
      <a
        class="c4 c5"
        href="https://jdmiguel.netlify.app/"
      >
        JDMIGUEL
      </a>
    </div>
  </div>
</footer>
`;

exports[`Component: Footer should render with veil 1`] = `
.c4 {
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  cursor: pointer;
  color: #FBA52B;
}

.c4 i {
  font-size: 0.8em;
  margin-right: 4px;
}

.c0 {
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  overflow: hidden;
}

.c1 {
  background-color: #F3F3F3;
  height: 170px;
  opacity: 0.6;
  -webkit-animation: hikGge 0.15s linear;
  animation: hikGge 0.15s linear;
}

.c2 {
  background-color: #660000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 20px 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #D8D8D8;
}

.c3:not(:first-of-type) {
  margin-left: 5px;
}

.c3:not(:last-of-type):after {
  content: ' | ';
  margin-left: 3px;
}

.c3:last-of-type {
  margin-right: 1px;
}

.c6 {
  font-weight: 300;
}

.c7 {
  font-weight: 300;
  margin-right: 3px;
}

@media only screen and (min-width:540px) {
  .c4 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }

  .c4:hover {
    color: #EC7E00;
  }
}

@media only screen and (max-width:330px) {
  .c5 {
    font-size: 0.85rem;
  }
}

@media only screen and (min-width:768px) {
  .c1 {
    height: 150px;
  }
}

@media only screen and (min-width:375px) {
  .c3:not(:first-of-type) {
    margin-left: 7px;
  }
}

@media only screen and (min-width:375px) {
  .c3:not(:last-of-type):after {
    margin-left: 5px;
  }
}

@media only screen and (min-width:375px) {
  .c3:last-of-type {
    margin-right: 3px;
  }
}

@media only screen and (max-width:330px) {
  .c3 {
    font-size: 0.85rem;
  }
}

@media only screen and (max-width:330px) {
  .c6 {
    font-size: 0.85rem;
  }
}

@media only screen and (max-width:330px) {
  .c7 {
    font-size: 0.85rem;
  }
}

<footer
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <a
        class="c4 c5"
        href="https://github.com/jdmiguel/foodublin"
      >
        GITHUB
      </a>
    </div>
    <div
      class="c3"
    >
      <p
        class="c6"
      >
        FOODUBLIN ©2020
      </p>
    </div>
    <div
      class="c3"
    >
      <p
        class="c7"
      >
        BY
      </p>
      <a
        class="c4 c5"
        href="https://jdmiguel.netlify.app/"
      >
        JDMIGUEL
      </a>
    </div>
  </div>
</footer>
`;
